### PR TITLE
Rename devicelab catalina tests

### PR DESCRIPTION
**Pull request by keyonghan**
This PR renames:
1) `smoke_catalina_hot_mode_dev_cycle_ios__benchmark` => `hot_mode_dev_cycle_ios__benchmark`
2) `smoke_catalina_start_up` => `flutter_gallery_mac__start_up` (we already have one test `flutter_gallery__start_up` for linux/android).

This is a one time for all change on the test names, but it will affect the benchmark data which depends on test names.

https://github.com/flutter/flutter/issues/73907

### CODE REVIEW

**Review by zanderso — APPROVED** [review]

File: .ci.yaml [hunk 1]
```diff
@@ -3011,14 +3011,14 @@ targets:
       task_name: service_extensions_test
     scheduler: luci
 
-  - name: Mac_android smoke_catalina_start_up
+  - name: Mac_android flutter_gallery_mac__start_up
```
Comment (CaseyHillers) [inline .ci.yaml:3014]: Based on https://github.com/flutter/flutter/wiki/Adding-a-new-Test-Shard#steps-to-add-a-new-framework-test-shard, should we be duplicating these with `bringup: true` to let the name soak in?

Comment (keyonghan) [reply .ci.yaml:3014]: Based on the above doc, shall we update the `framework_config.star` specifically, or simply update `.ci.yaml` with `bringup:true` to wait for the roll?

**Review by jmagman — APPROVED** [review]
Thanks, LGTM!

File: .ci.yaml (hunk 1 above)
Comment (CaseyHillers) [reply .ci.yaml:3014]: Good point :-)

Can you update the docs, and mention that renaming a target that is postsubmit only doesn't require following the bringup process?

**Review by CaseyHillers — APPROVED** [review]

File: .ci.yaml (hunk 1 above)
Comment (keyonghan) [reply .ci.yaml:3014]: Will update in a separate PR.

Comment (keyonghan) [reply .ci.yaml:3014]: Added a note at the bottom: https://github.com/flutter/flutter/wiki/Adding-a-new-Test-Shard

**Pull request merged**